shim/src/realm.js

```diff
@@ -44,27 +44,29 @@ const RealmRecForRealmInstance = new WeakMap();
 function getRealmRecForRealmInstance(realm) {
   if (Object(realm) !== realm) {
     // Detect non-objects.
-    throw new TypeError();
+    throw new TypeError('bad object, not a Realm instance');
```

Comment:
internal error?

Reply:
Also, can we really conclude "not a realm instance"? It seems that the message is more specific than the assertion.

Reply:
I think we hit that line if the `this` binding is unusual when the `evaluate` method is called (e.g. `r = Realm.makeRootRealm(); Reflect.apply(r.evaluate, 'this is weird', '1+2')`. So I think it'd require a deliberate action by the caller to get into that state.

A few lines later (where we check if the `this` is in the map at all): before I updated tests to stop using `new Realm`, they were hitting this case, so I figured it'd be a good idea to explicitly mention the non-constructor nature of Realm. But it could also be due to some internal error, or someone (I think) making an object that inherits from a real Realm and trying to call it.

But yeah, I'm happy to make that message less specific.

Reply:
hm, if it's not an Object then it's certainly not a Realm instance/constructor, so I suppose the message is technically accurate (the contrapositive is not true: being an Object doesn't automatically make it a Realm constructor, and we detect that by seeing whether it's in the map or not). Maybe we can think of the error message as instructions to ourselves? (if/when they get printed, it's because some other code we wrote failed to pass a constructor/instance in, so those are a hint that might help us with debugging)